File: core/repository/speculative_topology_engine.py
from __future__ import annotations

from typing import Any, Dict

from core.evidence.speculative_inference_engine import suppress_speculative_inference
from core.repository.topology_self_confirmation_engine import suppress_topology_self_confirmation
from core.repository.recursive_topology_lock_engine import suppress_recursive_topology_lock
from core.repository.civilization_topology_engine import apply_civilization_topology
from core.repository.anti_capture_topology_engine import apply_anti_capture_topology
from core.repository.sovereignty_topology_engine import apply_sovereignty_topology
from core.repository.openness_topology_engine import apply_openness_topology


def suppress_speculative_topology_edge(edge: Dict[str, Any]) -> Dict[str, Any]:
    ev = edge.get("evidence", []) or []
    spec = suppress_speculative_inference("topology_edge", ev, inferred=bool(edge.get("inferred")))
    suppressed = spec.get("suppressed", False)
    reality = {"aligned": bool(ev) and not suppressed, "parser_bounded": bool(ev)}
    base = {
        **edge,
        "reality_alignment": reality,
        "boundary_pressure": {"edge": suppressed or not ev},
        "stability": {"stable": bool(ev), "level": "high" if ev else "low"},
        "supported": {"stated": bool(ev)},
        "unsupported": {"edge": suppressed or not ev, "speculative": spec.get("suppressed", False)},
        "suppressed": {"edge": suppressed, "reason": spec.get("record", {}).get("reason") if spec.get("record") else None},
        "fragility": edge.get("fragility", {"level": "medium" if not ev else "low"}),
        "uncertainty": edge.get("uncertainty", {"visible": not ev}),
        "ambiguity": edge.get("ambiguity", {"visible": False}),
        "confidence_caps": edge.get("confidence_caps", edge.get("confidence_limits", {})),
        "evidence": ev,
        "speculative_suppression": spec.get("record"),
    }
    return apply_openness_topology(
        apply_sovereignty_topology(
            apply_anti_capture_topology(
                apply_civilization_topology(
                    suppress_recursive_topology_lock(suppress_topology_self_confirmation(base))
                )
            )
        )
    )

File: src/repository/speculativeTopologyEngine.ts
/**
 * Converted from Python: core/repository/speculative_topology_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { suppressSpeculativeInference } from "../evidence/speculativeInferenceEngine.js";
import { suppressTopologySelfConfirmation } from "./topologySelfConfirmationEngine.js";
import { suppressRecursiveTopologyLock } from "./recursiveTopologyLockEngine.js";
import { applyCivilizationTopology } from "./civilizationTopologyEngine.js";
import { applyAntiCaptureTopology } from "./antiCaptureTopologyEngine.js";
import { applySovereigntyTopology } from "./sovereigntyTopologyEngine.js";
import { applyOpennessTopology } from "./opennessTopologyEngine.js";

export function suppressSpeculativeTopologyEdge(edge: any): any {
  var ev: any = py.or2(py.get(edge, "evidence", []), () => ([]));
  var spec: any = suppressSpeculativeInference("topology_edge", ev, py.truthy(py.get(edge, "inferred")));
  var suppressed: any = py.get(spec, "suppressed", false);
  var reality: any = {"aligned": py.and2(py.truthy(ev), () => (!py.truthy(suppressed))), "parser_bounded": py.truthy(ev)};
  var base: any = {...(edge), "reality_alignment": reality, "boundary_pressure": {"edge": py.or2(suppressed, () => (!py.truthy(ev)))}, "stability": {"stable": py.truthy(ev), "level": (py.truthy(ev) ? "high" : "low")}, "supported": {"stated": py.truthy(ev)}, "unsupported": {"edge": py.or2(suppressed, () => (!py.truthy(ev))), "speculative": py.get(spec, "suppressed", false)}, "suppressed": {"edge": suppressed, "reason": (py.truthy(py.get(spec, "record")) ? py.get(py.get(spec, "record", {}), "reason") : null)}, "fragility": py.get(edge, "fragility", {"level": (!py.truthy(ev) ? "medium" : "low")}), "uncertainty": py.get(edge, "uncertainty", {"visible": !py.truthy(ev)}), "ambiguity": py.get(edge, "ambiguity", {"visible": false}), "confidence_caps": py.get(edge, "confidence_caps", py.get(edge, "confidence_limits", {})), "evidence": ev, "speculative_suppression": py.get(spec, "record")};
  return applyOpennessTopology(applySovereigntyTopology(applyAntiCaptureTopology(applyCivilizationTopology(suppressRecursiveTopologyLock(suppressTopologySelfConfirmation(base))))));
}
export { applyAntiCaptureTopology, applyCivilizationTopology, applyOpennessTopology, applySovereigntyTopology, suppressRecursiveTopologyLock, suppressSpeculativeInference, suppressTopologySelfConfirmation };
